GOVERNANCE.md

```diff
@@ -0,0 +1,147 @@
+
+The PR should only be opened no earlier than 6 weeks before the end of the project lead's term.
+The PR should be kept open for no less than 4 weeks. The PR can only be merged after the end of the
+last project lead's term, with more +1 than -1 in the binding votes.
+
+When there are conflicting PRs about changes in project lead, the PR with the most binding +1 votes is merged.
+
+The project lead can volunteer to step down.
+
+## Changes in Project Governance
+
+Changes in project governance (GOVERNANCE.md) could be initiated by opening a github PR.
+The PR should only be opened no earlier than 6 weeks before the end of the project lead's term.
+The PR should be kept open for no less than 4 weeks. The PR can only be merged follow the same
+voting process as in `Changes in Project Lead`.
+
+## Decision making process
+
+Decisions are build on consensus between maintainers.
+Proposals and ideas can either be submitted for agreement via a github issue or PR,
+or by sending an email to `maintainers@coredns.io`.
+
+In general, we prefer that technical issues and maintainer membership are amicably worked out between the persons involved.
+If a dispute cannot be decided independently, get a third-party maintainer (e.g. a mutual contact with some background
+on the issue, but not involved in the conflict) to intercede.
+If a dispute still cannot be decided, the project lead has the final say to decide an issue.
+
+Decision making process should be transparent to adhere to
+the principles of CoreDNS project.
+
+All proposals, ideas, and decisions by maintainers or the project lead
+should either be part of a github issue or PR, or be sent to `maintainers@coredns.io`.
+
+## Github Project Administration
+
+The __coredns__ GitHub project maintainers team reflects the list of Maintainers.
+
+## Other Projects
+
+The CoreDNS organization is open to receive new sub-projects under its umbrella. To accept a project
+into the __CoreDNS__ organization, it has to meet the following criteria:
+
+- Must be licensed under the terms of the Apache License v2.0
+- Must be related to one or more scopes of the CoreDNS ecosystem:
+  - CoreDNS project artifacts (website, deployments, CI, etc)
+  - External plugins
+  - Other DNS related processing
+- Must be supported by a Maintainer not associated or affiliated with the author(s) of the sub-projects
+
+The submission process starts as a Pull Request or Issue on the
+[coredns/coredns](https://github.com/coredns/coredns) repository with the required information
+mentioned above. Once a project is accepted, it's considered a __CNCF sub-project under the umbrella
+of CoreDNS__.
+
+## New Plugins
+
+The CoreDNS is open to receive new plugins as part of the CoreDNS repo. The submission process
+is the same as a Pull Request submission. Unlike small Pull Requests though, a new plugin submission
+should only be approved by a maintainer not associated or affiliated with the author(s) of the
```

Comment:
What precisely is meant by "associated or affiliated"?  We are all associated/affiliated via working on CoreDNS.

Does this mean "working for the same company/organization" ?

Reply:
I think the idea is to not including `the affiliation/association through coredns or cncf`. Don't know what is the best phase but happy to take any suggestions.

Reply:
How about ...

> ... should only be approved by a maintainer not in the same company/organization as the author(s) ...